exercises/poker/src/example/java/Hand.java

```diff
@@ -0,0 +1,106 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Hand {
+	private String input;
+	private int score;
+
+	public Hand(String hand) {
+		this.input = hand;
+		this.score = scoreHand(parseCards(hand));
+	}
+
+	public int getScore() {
+		return score;
+	}
+
+	public String getInput() {
+		return input;
+	}
+
+	private static List<Card> parseCards(String hand) {
+		ArrayList<Card> parsedCards = new ArrayList<Card>();
+		String[] parseString = hand.split(" ");
+		for (int i = 0; i < parseString.length; i++) {
+			parsedCards.add(new Card(parseString[i]));
+		}
+		return parsedCards;
+	}
+
+	private static Map<Integer, Integer> getFrequencyMap(List<Card> cards) {
+		Map<Integer, Integer> frequencymap = new HashMap<Integer, Integer>() ;
+		for (Card c : cards) {
+			if (frequencymap.containsKey(c.getRank())) {
+				frequencymap.put(c.getRank(), frequencymap.get(c.getRank()) + 1);
+			} else {
+				frequencymap.put(c.getRank(), 1);
+			}
+		}
+		return frequencymap;
+	}
+
+	private static int scoreHand(List<Card> cards) {
+		List<Card> cardsByRank = new ArrayList<Card>();
+
+		cardsByRank = cards.stream().sorted(Comparator.comparing(Card::getRank)).unordered()
+				.collect(Collectors.toList());
+
+		Map<Integer, Integer> frequencymap = getFrequencyMap(cards);
+		List<Integer> ranks = frequencymap.entrySet().stream().map(x -> x.getKey())
+				.sorted(Comparator.reverseOrder()).collect(Collectors.toList());
+
+		frequencymap = frequencymap.entrySet().stream()
+				.sorted(Map.Entry.comparingByValue(Collections.reverseOrder())).collect(Collectors
+						.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e1, LinkedHashMap::new));

```

Comment:
I use a LinkedHash because it maintains insertion order of elements. I know that it requires more memory, but I tried to use HashMap and TreeMap. Both didn't give me the right order.

I also tried your suggestions but i've go the same probleme that I've got with HashMap.
Sometime i've got the right order, sometime not. The twoSquares test won't work with Hash or your suggestions.
Example :
"2S 2H 2S 8D 2H";
"4S 5H 5S 5D 5H";

Before ordering the frequencyMap  :
{2=4, 8=1}
{4=1, 5=4}
After :
{2=4, 8=1}
{4=1, 5=4}

Nothing change ... It seems to use the key to order even with 
.sorted(Map.Entry.comparingByValue())
or 
.sorted(Map.Entry.comparingByValue(Collections.reverseOrder())
(i've got the same results btw, it's weird ....)

What i want is :
{2=4, 8=1}
{5=4, 4=1}

Do you have any suggestions ?

About getFrequencyMapInDescendingOrder, I had a reason for not doing this initially, but can't remembre. Anyway I change it and everything keeps working, so thanks !